bin/switchy

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python2.7
+# vim: tabstop=4 softtabstop=4 shiftwidth=4 textwidth=80 smarttab expandtab
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import sys
+import re
+import time
+import click
+import switchy
+import logging
+
+@click.group()
+def cli():
+    pass
+
+@cli.command('list-apps')
+def list_apps():
+    switchy.apps.load()
+    click.echo('Registered Switchy Apps:\n')
+    for app in switchy.apps.next():
+        click.echo('{} - {}'.format(app.name, app.help))
+
+@cli.command()
+@click.argument('file-name', nargs=1, required=True, type=click.Path(exists=True))
+def plot(file_name):
+    from switchy.apps.measure import metrics
+    m = metrics.load(file_name)
+    click.echo('Plotting {} ...\n'.format(file_name))
+    m.plot(block=True)
+
+@cli.command()
+@click.argument('slaves', nargs=-1, required=True)
+@click.option('--proxy', default=None, help='Hostname or IP address of the proxy device (this is usually the device you are testing)', required=True)
+@click.option('--profile', default='internal', help='Profile to use for outbound calls in the load slaves')
+@click.option('--rate', default=None, help='Call rate')
+@click.option('--limit', default=None, help='Maximum number of concurrent calls')
+@click.option('--max-offered', default=None, help='Maximum number of calls to place before stopping the program')
+@click.option('--duration', default=None, help='Duration of calls in seconds')
+@click.option('--interactive/--non-interactive', default=False, help='Whether to jump into an interactive session after setting up the call originator')
+@click.option('--debug/--no-debug', default=False, help='Whether to enable debugging')
+@click.option('--app', default='Bert', help='Switchy application to execute (see list-apps command to list available apps)')
+@click.option('--metrics-file', default=None, help='Store metrics at the given file location')
+def run(slaves, proxy, profile, rate, limit, max_offered, duration, interactive, debug, app, metrics_file):
+    logging.basicConfig()
+
+    # Check if the specified (or default) app is valid
+    switchy.apps.load()
+    appdata = switchy.apps.get(app)
+    if not appdata:
+        raise click.ClickException('Unknown app {}. Use list-apps command to list available apps'.format(app))
+
+    # TODO: get_originator() receives an apps tuple (defaults to Bert) to select the application
+    # we should accept --app multi argument list to set multiple apps
+    o = switchy.get_originator(slaves, apps=(appdata.cls, ))
+
+    # Set basic originator settings
+    if rate:
```

Comment:
You can actually pass in these args into the `get_originator` constructor since `**kwargs` are forwarded down to the instance.
See here: https://github.com/sangoma/switchy/blob/master/switchy/apps/call_gen.py#L56